lib/zold/score.rb

```diff
@@ -157,22 +161,18 @@ def reduced(max = 4)
 
     def next
       raise 'This score is not valid' unless valid?
-      idx = 0
-      loop do
-        suffix = idx.to_s(16)
-        score = Score.new(
-          time: time, host: host, port: port, invoice: invoice, suffixes: suffixes + [suffix],
-          strength: strength
-        )
-        return score if score.valid?
-        if score.expired?
-          return Score.new(
-            time: Time.now, host: host, port: port, invoice: invoice,
-            suffixes: [], strength: strength
-          )
-        end
-        idx += 1
-      end
+      return Score.new(
+        time: Time.now, host: host, port: port, invoice: invoice,
+        suffixes: [], strength: strength
+      ) if expired?
+      idx = ScoreExt.calculate_nonce_multi_core(
+        CORES, "#{(suffixes.empty?) ? prefix : hash} ", strength
+      )
+      suffix = idx.to_s(16)
+      return Score.new(
```

Comment:
In general, I prefer to have a single point of return in methods, otherwise it is not very easy at first glance what are we returning

Reply:
Logic over style here as well.